File: dataset/output/OPENAI_gpt-4o/Task123_OPENAI_gpt-4o.java
public class UserSession {
    private int userId;
    private boolean active;

    public UserSession(int userId) {
        this.userId = userId;
        this.active = true;
    }

    public void endSession() {
        this.active = false;
    }

    public int getUserId() {
        return userId;
    }

    public boolean isActive() {
        return active;
    }

    public static void main(String[] args) {
        UserSession session = new UserSession(1);
        System.out.println(session.getUserId());
        System.out.println(session.isActive());
        session.endSession();
        System.out.println(session.isActive());
    }
}

File: dataset/output/OPENAI_gpt-4o/Task123_OPENAI_gpt-4o.cpp
#include <iostream>
#include <chrono>
#include <thread>

class UserSession {
    std::string userId;
    std::chrono::time_point<std::chrono::steady_clock> startTime;

public:
    UserSession(const std::string& userId) : userId(userId), startTime(std::chrono::steady_clock::now()) {}

    double getDuration() {
        auto currentTime = std::chrono::steady_clock::now();
        std::chrono::duration<double> duration = currentTime - startTime;
        return duration.count();
    }

    std::string getUserId() {
        return userId;
    }
};

int main() {
    UserSession session("User1");
    std::this_thread::sleep_for(std::chrono::seconds(2));
    std::cout << "CPP: Session duration for " << session.getUserId() << ": " << session.getDuration() << " seconds" << std::endl;
    return 0;
}